src/ui/components/filters/filterboxcomponent.js

```diff
@@ -275,14 +275,16 @@ export default class FilterBoxComponent extends Component {
   /**
    * Save current filters to storage to be used in the next search
    * @private
+   * @param {boolean} replaceHistory Whether we replace or push a new history
+   *                                 state for the associated changes
    */
-  _saveFilterNodesToStorage () {
+  _saveFilterNodesToStorage (replaceHistory) {
```

Comment:
instead of using this method directly, would it make sense to have two separate methods that call this one? one for when the component is mounted and the other for updates? Otherwise it's less clear to me why do some calls replaceHistory and others push it? Or maybe instead of calling this replaceHistory, which leaks implementation details, you could call it something like isNewSearch?

Reply:
I don't know if there are ideal solutions here. We need to get this out for Mercy and I don't think we're gonna come up with something we all feel good about in the time allotted. The `Facets` and `Filters` components are just a big mess. I'm not blaming any of us for that. They were in bad shape when we inherited them. Over the past couple months, we've been repeatedly asked to quickly patch things for Mercy. This has made the situation worse. It's clear that when these components were designed, there was not much though put into how they'd work with refreshes, back nav, etc. 

I'm advocating that we just get something working for them (that doesn't add regressions or other bugs) and plan to rebuild all of this soon.